src/CMakeLists.txt

```diff
@@ -9,6 +8,66 @@ include_directories("md/inc")
 include_directories("classlibnative/bcltype")
 include_directories("classlibnative/cryptography")
 include_directories("classlibnative/inc")
+include_directories("${GENERATED_INCLUDE_DIR}")
+
+set (ScriptGeneratedEventFiles
+      ${GENERATED_INCLUDE_DIR}/clretwallmain.h
+      ${GENERATED_INCLUDE_DIR}/etmdummy.h
+    )
+set (GeneratedEventFiles)
+
+if(WIN32)
```

Comment:
I am sorry, I guess I must have confused you regarding what I was intending to achieve. What I was after was to make it shorter. The custom target is for handling the header file dependencies, there is no need to use it for the .rc and for the ClrEtwAll.h which is not used anywhere in the codebase as a dependency. So e.g. this:
```cmake
if(WIN32)
  set (GenEventFilesScript "${CLR_DIR}/src/scripts/genWinEtw.py")
    add_custom_command(
      COMMENT "Generating ETW resource Files"
      COMMAND ${MC} -h ${GENERATED_INCLUDE_DIR} -r ${GENERATED_INCLUDE_DIR} -b -co -um -p FireEtw "${VM_DIR}/ClrEtwAll.man"
      OUTPUT ${GENERATED_INCLUDE_DIR}/ClrEtwAll.h
      OUTPUT ${GENERATED_INCLUDE_DIR}/ClrEtwAll.rc
      DEPENDS "${VM_DIR}/ClrEtwAll.man"
       )
else()
  set (GenEventFilesScript "${CLR_DIR}/src/scripts/genXplatEventing.py")
endif()

set (ScriptGeneratedEventFiles
      ${GENERATED_INCLUDE_DIR}/clretwallmain.h
      ${GENERATED_INCLUDE_DIR}/etmdummy.h
    )

add_custom_command(
  COMMENT "Generating Eventing Files"
  COMMAND ${PYTHON} -B -Wall -Werror ${GenEventFilesScript} ${GenEventArgs} --man "${VM_DIR}/ClrEtwAll.man" --exc "${VM_DIR}/ClrEtwAllMeta.lst" --dummy "${GENERATED_INCLUDE_DIR}/etmdummy.h"
  OUTPUT ${ScriptGeneratedEventFiles}
  DEPENDS ${GenEventFilesScript} "${VM_DIR}/ClrEtwAll.man" "${VM_DIR}/ClrEtwAllMeta.lst" "${CLR_DIR}/src/scripts/genXplatEventing.py"
    )

add_custom_target(
  GeneratedEventingFiles
  DEPENDS ${ScriptGeneratedEventFiles}
)

```